docs/src/glossary.md

```diff
@@ -0,0 +1,81 @@
+# ChainRules Glossary
+
+This glossary serves as a quick reference for common terms used in the field of Automatic Differentiation, as well as those used throughout the documentation relating specifically to ChainRules.
+
+##Definitions:
+
+###Adjoint:
+
+The conjugate transpose of the Jacobian for a given function `f`.
+
+###Derivative:
+
+The derivative of a function `y = f(x)` with respect to the independent variable `x` denoted `f'(x)` or `dy/dx` is the rate of change of the dependent variable `y` with respect to the change of the independent variable `x`. In multiple dimensions, we may refer to the gradient of a function.
```

Comment:
Correct me if I am wrong, but should
```
the rate of change of the dependent variable `y` with respect to the change of the independent variable `x`.
```

be simply changed to
```
the rate of change of the dependent variable `y` with respect to the independent variable `x`.
```

Reply:
Maybe something like this is more appropriate:

"The derivative of a function `y = f(x)` with respect to the independent variable `x` denoted `f'(x)` or `dy/dx` is the measure of the rate of change of the dependent variable `y` with respect to the rate of change of the independent variable `x`. In multiple dimensions, we may refer to the gradient of a function."

I agree that it is normally referred to "with respect to x", however I think it is important to explain what is being measured. Would the changes above seem suitable?

Reply:
>  In multiple dimensions, we may refer to the gradient of a function."

Maybe better to say:
in multiple dimensions the derivative is not defined.
Instead there are the partial derivatives, the directional derivative, the jacobian (called gradient for scalar-valued functions).

Reply:
Updated